source/common/common/utility.cc

```diff
@@ -122,6 +122,15 @@ std::vector<std::string> StringUtil::split(const std::string& source, const std:
   return ret;
 }
 
+std::string StringUtil::join(const std::vector<std::string>& source, const std::string& delimiter) {
```

Comment:
Might be able to templatize and use `std::copy` to simplify this, see https://stackoverflow.com/questions/1430757/c-vector-to-string.